src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -142,12 +147,55 @@ void primitive_inst::realloc_if_needed() {
         _output = _network.get_engine().reinterpret_buffer(*_output, _node.get_output_layout());
     }
 }
+std::string primitive_inst::get_layout_key() {
+    std::string layout_key_str = "";
+    if (!_node.is_valid_output_layout()) {
+        std::cout << id() << " has invaild output layout!" << std::endl;
+        return "";
+    }
+    layout_key_str = id() + "_" + std::to_string(_node.get_unique_id());
+
+    auto data_tensor = convert_data_tensor(_node.get_output_layout());
+    layout_key_str += "_" + kernel_selector::toString_v2(data_tensor);
+
+    for (auto in : _node.get_dependencies()) {
+        if (!in->is_constant()) {
+            auto data_tensor = convert_data_tensor(in->get_output_layout());
+            layout_key_str += "_" + kernel_selector::toString_v2(data_tensor);
+        }
+    }
+    return layout_key_str;
+}
 
 void primitive_inst::update_impl() {
+    auto pid = id();
     if (!_node.is_type<data>() && !(_node.is_type<mutable_data>() && _node.get_dependencies().empty())) {
-        PRINT_TIME(_impl = std::move(_node.type()->choose_impl(_node)));
-        PRINT_TIME(_network.get_program()->compile());
-        PRINT_TIME(_impl->init_kernels());
+        auto layout_key = get_layout_key();
+        auto ret = std::getenv("NO_USE_IMPL_CACHE");
+        if ((ret == nullptr) && layout_key != "") {
+            auto cache = _network.get_program()->get_primitive_impl_cache();
+            bool is_hitted = false;
```

Comment:
nit: hit's past participle is just "hit" too, so .. rename the variable as is_hit ?